lib/rules/function-url-quotes/index.js

```diff
@@ -44,96 +46,78 @@ const rule = (primary, secondaryOptions, context) => {
 			return;
 		}
 
+		const exceptEmpty = optionsMatches(secondaryOptions, 'except', 'empty');
+		const emptyArgumentPatterns = new Set(['', "''", '""']);
+
 		root.walkAtRules(checkAtRuleParams);
 		root.walkDecls(checkDeclParams);
 
 		/**
 		 * @param {import('postcss').Declaration} decl
 		 */
 		function checkDeclParams(decl) {
-			functionArgumentsSearch(decl.toString(), /^url$/i, (args, index) => {
-				checkArgs(args, decl, index, 'url');
+			const value = getDeclarationValue(decl);
+			const startIndex = declarationValueIndex(decl);
+			const parsed = functionArgumentsSearch(value, /^url$/i, (args, index, funcNode) => {
+				checkArgs(decl, args, startIndex + index, funcNode);
 			});
+
+			if (context.fix) {
+				decl.value = parsed.toString();
+			}
 		}
 
 		/**
 		 * @param {import('postcss').AtRule} atRule
 		 */
 		function checkAtRuleParams(atRule) {
-			functionArgumentsSearch(atRule.params, /^url$/i, (args, index) => {
-				checkArgs(args, atRule, index + atRuleParamIndex(atRule), 'url');
-			});
-			functionArgumentsSearch(atRule.params, /^url-prefix$/i, (args, index) => {
-				checkArgs(args, atRule, index + atRuleParamIndex(atRule), 'url-prefix');
-			});
-			functionArgumentsSearch(atRule.params, /^domain$/i, (args, index) => {
-				checkArgs(args, atRule, index + atRuleParamIndex(atRule), 'domain');
-			});
-		}
-
-		/**
-		 * @param {import('postcss').AtRule} node
-		 * @param {string} args
-		 * @param {number} index
-		 */
-		function addQuotesForAtRule(node, args, index) {
-			const fixedName = `"${args}"`;
-			const openIndex = index - atRuleParamIndex(node);
-			const closeIndex = openIndex + args.length;
-
-			node.params =
-				node.params.substring(0, openIndex) + fixedName + node.params.substring(closeIndex);
-		}
-
-		/**
-		 * @param {import('postcss').Declaration} node
-		 * @param {string} args
-		 * @param {number} index
-		 */
-		function addQuotesForDecl(node, args, index) {
-			const fixedName = `"${args}"`;
-			const openIndex = index - declarationValueIndex(node);
-			const closeIndex = openIndex + args.length;
+			const params = getAtRuleParams(atRule);
+			const startIndex = atRuleParamIndex(atRule);
+			const parsed = functionArgumentsSearch(
+				params,
+				/^(url|url-prefix|domain)$/i,
```

Comment:
```suggestion
				/^url$/i,
```

> so it seems safe to remove the function from the test cases.

And from the rule itself. Same for [`domain()`](https://csstree.github.io/docs/syntax/#Function:domain).

Reply:
Fixed via 9a103c1a